File: TheRayCode/JavaScript/Creational/Builder/Example1/ConcreteBuilder.js
const Builder = require('./Builder');
const Product = require('./Product');

class ConcreteBuilder extends Builder {
    constructor() {
        super();
        this.product = new Product();
    }
    buildPartA() {
        this.product.add("PartA");
    }
    buildPartB() {
        this.product.add("PartB");
    }
    getResult() {
        return this.product;
    }
}
module.exports = ConcreteBuilder;

File: Python/Creational/Builder/Example1/Main/src/ConcreteBuilder.java
public class ConcreteBuilder implements Builder{
    private Product product = new Product();

    @Override
    public void buildPartA() {
        product.add("PartA");
    }

    @Override
    public void buildPartB() {
        product.add("PartB");
    }

    @Override
    public Product getResult() {
        return product;
    }
}
